pkg/sql/catalog/resolver/resolver.go

```diff
@@ -274,6 +274,56 @@ func ResolveSchemaNameByID(
 	return "", errors.Newf("unable to resolve schema id %d for db %d", schemaID, dbID)
 }
 
+// ResolveSchemaDescByID resolves a schema based on it's ID.
+func ResolveSchemaDescByID(
```

Comment:
I wouldn't call this `ResolveSchemaDescByID` because we aren't gauranteed to return a descriptor always. Just `ResolveSchemaByID` will do. Add some TODO's here for lucy and I to move this into the `descs.Collection`